job-service/src/main/java/com/hpe/caf/services/job/api/JobsResume.java

```diff
@@ -56,14 +72,78 @@ public static void resumeJob(final String partitionId, String jobId) throws Exce
             //  Get database helper instance.
             DatabaseHelper databaseHelper = new DatabaseHelper(config);
 
+            // Store the job status before resuming the job.
+            final Job.StatusEnum jobStatusBeforeResuming = databaseHelper.getJobStatus(partitionId, jobId);
+
             //  Resume the specified job.
-            LOG.debug("resume: Resuming the job...");
+            LOG.debug("resume: Updating the status of the job in the database...");
             databaseHelper.resumeJob(partitionId, jobId);
 
+            // Send a resume job message only if the job was paused (we allow resuming an active job, which is why we need to check this).
+            if (jobStatusBeforeResuming == Job.StatusEnum.PAUSED) {
```

Comment:
There is a race condition here that can between lines 76 and 83, if another request came in that caused the status to change. 

I think it is ok if it did happen, as the newTask Unstowing worker would get a resume job message, not be able to find any matching tasks in the db (because the job is active/complete whatever), and can either just ignore the message or log a warning "Received a job resume message but unable to find any tasks to be resumed"